sgxs-tools/src/sgx_detect/tests/mod.rs

```diff
@@ -784,8 +785,110 @@ impl Dependency<RunEnclaveProdWl> for RunEnclaveProd {
     const CONTROL_VISIBILITY: bool = true;
 }
 
+// EnclaveOS requirements
+#[derive(Default, DebugSupport, Update)]
+struct Eor {
+    graphene: Status,
+    node_agent: Status,
+    perm_daemon: Status,
```

Comment:
Only the gsgx device is truly an EnclaveOS requirement. The others are Data Shield requirements, or more precisely:

node_agent is a requirement to run Enclave Manager-enabled applications

perm_daemon is a "run containers that use SGX under a container runtime that limits device access with cgroups" requirement (meaning, not required for plain docker, but usually required for kubernetes)

This distinction may not matter for the time being, but noting it. @banne01 please chime in if you have thoughts.

Reply:
I want to resolve this question before merging

Reply:
@aandyl How should we label this? Should we use DataShield instead of EnclaveOS here?

Reply:
Given that perm daemon should be going away soon, let's overload perm daemon on EnclaveOS for now. I would propose the following changes:
* remove the EnclaveOS pre-install mode. just have one EnclaveOS mode that includes gsgx and perm daemon checks but not node agent checks
* add Enclave Manager mode, that checks the node agent
* maybe add a Data Shield command line argument that activates both EOS and EM modes.